### ISSUE TITLE
Indicate inputs of operations as files

### ISSUE THREAD

**Issue by forman** [CCI-Tools/cate-core#137]
We need to indicate whether an input of an operation is a file and which format is expected so that we can provide the right editors in the GUI. We need

* `file_open_mode` one of "r", "w", "rw"
* `file_filters` array of Dict(name: str, extensions: List[str])

**Comment by JanisGailis** [comment]
@forman 

What will be the interface on ```ops``` side? Something like this?:
```python
@op
@op_input(file, cate_type='file', file_mode='r', file_filters={'Text': ['md', 'txt']})
def file_op(ds: xr.Dataset, file: str):
    pass
```

By the way, I'd like to do something like ```cate_type```, as that will probably be workable for the solution of #130 #46

**Comment by JanisGailis** [comment]
This could be a specific 'cate type' Something that necessarily contains a tuple of strings and a dict.

**Comment by forman** [comment]
Already implemented in 0.7
